Cython/TestUtils.py

```diff
@@ -228,21 +229,42 @@ def unpack_source_tree(tree_file, workdir, cython_root):
     return workdir, header
 
 
+def fresh_cythonize(*args, **kwargs):
+    """Clear function caches and run cythonize"""
+    clear_function_and_Dependencies_caches()
+    Dependencies.cythonize(*args, **kwargs)
+
+
+def clear_function_and_Dependencies_caches():
+    Utils.clear_function_caches()
+    Dependencies._dep_tree = None  # discard method caches
+
+
 def write_file(file_path, content, dedent=False):
```

Comment:
`content` in `bytes` + `dedent = True` are incompatible and therefore `write_file(<path>, bytes(), dedent=True)` will raise an exception: `TypeError: cannot use a string pattern on a bytes-like object` in `dedent`. The same applies to `write_newer_file`.

Reply:
That's fine. Users will notice that when they try it.